### PR TITLE
chore(inputs.x509_cert): Cleanup code

### PR DESCRIPTION
**Pull request by srebhan**
- [ ] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

This PR cleans the code of certificate checking to add new features later.

### CODE REVIEW

File: plugins/inputs/x509_cert/x509_cert.go [hunk 1]
```diff
@@ -37,12 +38,118 @@ type X509Cert struct {
...
+
+func (c *X509Cert) Init() error {
+	// Check if we do have at least one source
+	if len(c.Sources) == 0 {
+		return errors.New("no source configured")
+	}
+
+	// Check the server name and transfer it if necessary
+	if c.ClientConfig.ServerName != "" && c.ServerName != "" {
+		return fmt.Errorf("both server_name (%q) and tls_server_name (%q) are set, but they are mutually exclusive", c.ServerName, c.ClientConfig.ServerName)
```
Comment (powersj) [inline plugins/inputs/x509_cert/x509_cert.go:62]: This previously was not an error right?

**Review by powersj — COMMENTED** [review]
one question about change in behavior otherwise, thanks!

File: plugins/inputs/x509_cert/x509_cert.go (hunk 1 above)
Comment (srebhan) [reply plugins/inputs/x509_cert/x509_cert.go:62]: It was an error before, but checked in a different (now dissolved) [function](https://github.com/influxdata/telegraf/blob/6cfae2a1fd898cf7dfbca98565bcb534bf671050/plugins/inputs/x509_cert/x509_cert.go#L73).

Comment (powersj) [reply plugins/inputs/x509_cert/x509_cert.go:62]: ah thank you

**Review by powersj — APPROVED** [review]

**Pull request merged**